aten/src/ATen/core/TensorBase.h

```diff
@@ -620,7 +620,13 @@ class TORCH_API TensorBase {
   TensorAccessor<T,N> accessor() const& {
     static_assert(N > 0, "accessor is used for indexing tensor, for scalars use *data_ptr<T>()");
     TORCH_CHECK(dim() == N, "TensorAccessor expected ", N, " dims but tensor has ", dim());
-    return TensorAccessor<T,N>(data_ptr<T>(),sizes().data(),strides().data());
+    T* ptr = nullptr;
+    if constexpr (std::is_const<T>::value) {
+      ptr = const_data_ptr<std::remove_const_t<T>>();
```

Comment:
IDK, should `const_data_ptr` do this removal automatically?

Reply:
Yeah that seems reasonable